data_structures/src/mainnet_validations.rs

```diff
@@ -209,6 +209,32 @@ pub fn after_second_hard_fork(epoch: Epoch, environment: Environment) -> bool {
     epoch >= SECOND_HARD_FORK && Environment::Mainnet == environment
 }
 
+/// Allows to check the active Witnet Improvement Proposals
+#[derive(Clone, Debug)]
+pub struct ActiveWips {
+    pub active_wips: HashMap<String, Epoch>,
+    pub block_epoch: Epoch,
+    pub environment: Environment,
+}
+
+impl ActiveWips {
+    pub fn first_hard_fork(&self) -> bool {
+        after_first_hard_fork(self.block_epoch, self.environment)
+    }
+
+    pub fn second_hard_fork(&self) -> bool {
+        after_second_hard_fork(self.block_epoch, self.environment)
+    }
+
+    pub fn wip0014(&self) -> bool {
+        // TODO: ignore environment?
```

Comment:
This is an interesting point. Different Witnet networks (e.g. mainnet vs. testnet) may activate an improvement or fail to do it at different times.

This should be a no-problem under the assumption that the activation state of an improvement is always derived from the local chain, and that nodes can't jump from one network to another without a full resync.

Reply:
I would expect that after an improvement proposal is enabled in mainnet any local network will start with that improvement proposal enabled. So we could hardcode a list of enabled WIPs somewhere, or since this is only used for developing we can even hardcode them in the code.

Reply:
That makes sense, given that WIP-0014§(17) states that:
> Once a protocol improvement is activated through this procedure, the set of protocol rules that it
introduces become part of the core Witnet protocol.